client/state/sites/hooks/use-premium-global-styles.ts

```diff
@@ -0,0 +1,78 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import wpcom from 'calypso/lib/wp';
+import { getSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type GlobalStylesInfo = {
+	shouldLimitGlobalStyles: boolean;
+	globalStylesInUse: boolean;
+};
+
+type Site = {
+	isCreated: boolean;
+	id: number;
+};
+
+const NEW_SITE_GLOBAL_STYLES_INFO: GlobalStylesInfo = {
+	// The next line should be replaced with true once the Gating global styles feature is live.
+	// That will make all non-created sites to limit global styles.
+	shouldLimitGlobalStyles: isEnabled( 'limit-global-styles' ),
+	globalStylesInUse: false,
+};
+
+// While we are loading the Global Styles Info we can't assume that we should limit global styles, or we would be
+// showing notices for paid sites until we fetch the data from the server.
+const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesInfo = {
+	shouldLimitGlobalStyles: false,
+	globalStylesInUse: false,
+};
+
+const getGSInfoEndpointForSite = ( siteId: number ) => ( {
+	path: `sites/${ siteId }/global-styles-info`,
+	apiNamespace: 'wpcom/v2/',
+} );
+
+const mapSiteIdToSite = ( siteId: number | null ) => ( {
+	isCreated: !! siteId,
+	id: siteId ?? 0,
+} );
+
+const getGlobalStylesInfoForSite = ( site: Site ) => {
+	if ( ! site.isCreated ) {
+		return NEW_SITE_GLOBAL_STYLES_INFO;
+	}
```

Comment:
The only case where we are going to need to check the feature flag is when sites aren't created. We'll need to revisit this file in the future so that we can replace the feature check with true, since every single site that isn't created will need to display any notices that should be displayed.